Scripts/Runtime/Entities/TaskDriver/AbstractTaskDriver.cs

```diff
@@ -416,11 +420,53 @@ void ITaskSetOwner.AddResolvableDataStreamsTo(Type type, List<AbstractDataStream
         {
             TaskSet.AddResolvableDataStreamsTo(type, dataStreams);
         }
+        
+        //*************************************************************************************************************
+        // MIGRATION
+        //*************************************************************************************************************
 
+        internal void AddToMigrationLookup(
```

Comment:
No because you won't have created all your Data yet. The Register call happens in the base constructor before you have a had a chance to run your derived constructor which creates all the Data Streams.

Reply:
oh, that's a difficult pattern where you have to make sure everything is declared before calling register. hmmm, fine for now but we should look to improve later.

Reply:
If you had to call Register I would agree for sure. 

But because you can't control the order I feel like it's ok? 

1. Your base constructor runs which automatically registers your instance to the system. 
2. Your custom constructor runs which lets you add all your data as you see fit. 
3. TaskDriverManagementSystem hardens all TaskDrivers on first OnStartRunning which makes sure everyone is aware of all data, jobs, migration etc.

Reply:
oh I understand now. I didn't quite get the order of operations. Yea that's good!